app/src/main/java/fr/neamar/kiss/MainActivity.java

```diff
@@ -377,6 +379,21 @@ public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
             searchEditText.requestFocus();
         }
 
+        // Pasting shared text from Sharesheet via intent-filter into kiss search bar
+        Intent receivedIntent = getIntent();
+        String receivedIntentAction = receivedIntent.getAction();
+        String receivedIntentType = receivedIntent.getType();
+        if (Intent.ACTION_SEND.equals(action) && receivedIntentType != null && "text/plain".equals(receivedIntentType)) {
```

Comment:
Check for `receivedIntentType != null` can be removed. It's not necessary because of next one.